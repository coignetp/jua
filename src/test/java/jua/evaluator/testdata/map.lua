x = {1, 2, 3}
print(x == {1, 2, 3})

function inspect(count)
    print("inspect number: " .. count)
    print(x[0])
    print(x[1])
    print(x[2])
    print(x[3])
    print(x[4])
    print(x[5])
    print(x[100])
    print(x['test'])
    print(x['test2'])
    print(#x)
    print("---")
end
inspect(1)

x[1] = 'hello'
inspect(2)

x['test'] = 'test-string'
inspect(3)

x.test = 'cc'
inspect(4)

x.test2 = 'new'
inspect(5)

table.remove(x, 1)
inspect(6)

table.insert(x, "last")
inspect(7)

x[#x + 1] = "new last"
inspect(8)

print(table.concat(x, "|"))

x.func = function() print("hi") end
x.func()

x = {1, 2, 3}
inspect(9)

x[100] = 100
inspect(10)

x[5] = 5
inspect(11)

x[4] = 4
inspect(12)

print(#{hello = 1, hi = 2})
print(#{[4.5] = 1, [3.2] = 2})
