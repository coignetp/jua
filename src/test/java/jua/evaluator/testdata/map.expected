false
inspect number: 1
nil
1
2
3
nil
nil
nil
nil
nil
3
---
inspect number: 2
nil
hello
2
3
nil
nil
nil
nil
nil
3
---
inspect number: 3
nil
hello
2
3
nil
nil
nil
test-string
nil
3
---
inspect number: 4
nil
hello
2
3
nil
nil
nil
cc
nil
3
---
inspect number: 5
nil
hello
2
3
nil
nil
nil
cc
new
3
---
inspect number: 6
nil
2
3
nil
nil
nil
nil
cc
new
2
---
inspect number: 7
nil
2
3
last
nil
nil
nil
cc
new
3
---
inspect number: 8
nil
2
3
last
new last
nil
nil
cc
new
4
---
2|3|last|new last
hi
inspect number: 9
nil
1
2
3
nil
nil
nil
nil
nil
3
---
inspect number: 10
nil
1
2
3
nil
nil
100
nil
nil
3
---
inspect number: 11
nil
1
2
3
nil
5
100
nil
nil
3
---
inspect number: 12
nil
1
2
3
4
5
100
nil
nil
5
---
0
0
1
1
inspect number: 13
nil
1
2
3
4
5
100
nil
nil
5
---
